packages/lit-dev-content/site/tutorials/content/working-with-lists/01.md

```diff
@@ -0,0 +1,66 @@
+While Lit allows rendering values straight out of an iterable, it would be nice
+to have the ability to apply some transformation like adding markup for each
+item. Lit provides a built-in `map` directive that works on any iterable such as
+arrays, sets, maps, and even generators. It returns an iterable containing the
+result of calling the provided callback function on each item.
+
+In this example, a custom element is presented with a property named `items`
+that contains a set of strings.
+
+Render each item in the set as a list item (`<li>`) containing that string.
+Begin by importing the `map` directive.
+
+{% switchable-sample %}
+
+```ts
+// my-element.ts
+import {map} from 'lit/directives/map.js';
+```
+
+```js
+// my-element.js
+import {map} from 'lit/directives/map.js';
+```
+
+{% endswitchable-sample %}
+
+Use the directive in the rendered template, providing a callback function that
+returns a template wrapping each item with a list item (`<li></li>`) tag, and
+place it within the unordered list (`<ul></li>`) of the component's template.
+
+{% switchable-sample %}
+
+```ts
```

Comment:
Do we need separate ts/js samples when they're exactly the same? @aomarks

Reply:
For most of these, I've added separate samples because of the first line comment that names the file and wanted the file extensions to match the selected language. I had a comment about adding the file name for where the code snippet should go, though arguably since there's only ever 1 file you're editing for this tutorial it could be omitted.

Reply:
I'd say it's okay to omit the switchable-sample macro in these cases

Reply:
Should I do
```
// my-element.(js|ts)
```
for the first line comment in the samples then, or omit the filename comment all-together for this tutorial since the reader is only ever working on the one file?

Maybe this is another thing that we'd want a consistency pass across all tutorials as well. @arthurevans what do you think?

Reply:
Oh... that's enough of a difference to keep the two then. I missed that and thought they were identical.